File: webapi/Models/System/SysInsuranceType.cs
using System.ComponentModel.DataAnnotations;

namespace webapi.Models.System
{
    public record SysInsuranceType
    {
        [Key]
        public int Id { get; set; }
        [MaxLength(2)]
        public required string Code { get; set; }
        public required string InsuranceType { get; set; }
        public required decimal[] HealthInsuranceArticle40 { get; set; }
        public required decimal[] DooWithouthTzpbInsurer { get; set; }
        public required decimal[] DooWithouthTzpbEmployee { get; set; }
        public required decimal[] HealthInsuranceInsurer { get; set; }
        public required decimal[] HealthInsuranceEmployee { get; set; }
        public required decimal[] TeacherPensionFund { get; set; }
        public required decimal[] ProfessionalPensionFund { get; set; }
        public required decimal[] UniversalPensionInsurer { get; set; }
        public required decimal[] UniversalPensionEmployee { get; set; }
        public required decimal[] HealthInsurance { get; set; }
        public decimal GvrcFund { get; set; }
        public decimal DodTax {get; set; }
    }
}

File: reactapp/src/models/System/SysInsuranceType.ts
export interface SysInsuranceType {
    id: int,
    code: String,
    insuranceType: String,
    healthInsuranceArticle40: Array<String>,
    dooWithouthTzpbInsurer: Array<String>,
    dooWithouthTzpbEmployee: Array<String>,
    healthInsuranceInsurer: Array<String>,
    healthInsuranceEmployee: Array<String>,
    teacherPensionFund: Array<String>,
    professionalPensionFund: Array<String>,
    universalPensionInsurer: Array<String>,
    universalPensionEmployee: Array<String>,
    healthInsurance: Array<String>,
    gvrcFund: String,
}